### ISSUE TITLE
Generate api endpoints from mattermost openapi doc

### ISSUE THREAD

**Issue by Vaelor** [Vaelor/python-mattermost-driver#60]
Instead of manually adding each endpoint, a script to generate all endpoints from the openapi documentation would be nice.

This could also add more infos to every endpoint instead of only a `options` or `params` parameter.

**Comment by rittelle** [comment]
Generators like [openapi generator](https://github.com/OpenAPITools/openapi-generator) or [autorest](https://github.com/Azure/autorest) seem to be a good thing to consider.
But I haven't been able to make either of them work with the Mattermost openapi spec, yet …

**Comment by Vaelor** [comment]
Yeah, I have to work with stuff like that at work, so I should be able to do that, I just never really have the time for it. But I guess I am quite a bit behind on adding endpoints anyway so I should probably finally have a look into that.

**Comment by rittelle** [comment]
If I were able to get one at least one of them to work, I'd give an implementation a shot. I eventually gave up (too many other things to do ^^ ).
